File: Leetcode/10_array_partition-1.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var arrayPairSum = function (nums) {
  return nums
    .sort((a, b) => a - b)
    .reduce((p, c, i) => {
      return i % 2 === 0 ? p + c : p;
    }, 0);
};

console.log(arrayPairSum([1, 4, 2, 3]));

File: Leetcode/10_array_partition-1.py
# https://leetcode.com/problems/array-partition-i/https://leetcode.com/problems/array-partition-i/
from typing import List


def arrayPairSum(nums: List[int]) -> int:
    return sum(sorted(nums)[::2])


print(arrayPairSum([1, 4, 3, 2]))

